pkg/pool-weighted/contracts/smart/ManagedPool.sol

```diff
@@ -435,30 +448,118 @@ contract ManagedPool is BaseWeightedPool, ReentrancyGuard {
         SwapRequest memory swapRequest,
         uint256 currentBalanceTokenIn,
         uint256 currentBalanceTokenOut
-    ) internal override returns (uint256) {
+    ) internal virtual override whenNotPaused returns (uint256) {
+        // Swaps are disabled while the contract is paused.
         _require(getSwapEnabled(), Errors.SWAPS_DISABLED);
 
-        return super._onSwapGivenIn(swapRequest, currentBalanceTokenIn, currentBalanceTokenOut);
+        (uint256[] memory normalizedWeights, uint256[] memory preSwapBalances) = _getWeightsAndPreSwapBalances(
+            swapRequest,
+            currentBalanceTokenIn,
+            currentBalanceTokenOut
+        );
+
+        // balances (and swapRequest.amount) are already upscaled by BaseMinimalSwapInfoPool.onSwap
+        uint256 amountOut = super._onSwapGivenIn(swapRequest, currentBalanceTokenIn, currentBalanceTokenOut);
```

Comment:
I think the function you want to override is not `onSwapGivenIn` and `onSwapGivenOut`, but `onSwap`. `onSwapGivenIn` and out are fully unaware of swap fees. From their docs:

```
/*
    * @dev Called when a swap with the Pool occurs, where the amount of tokens entering the Pool is known.
    *
    * Returns the amount of tokens that will be taken from the Pool in return.
    *
    * All amounts inside `swapRequest`, `balanceTokenIn` and `balanceTokenOut` are upscaled. The swap fee has already
    * been deducted from `swapRequest.amount`.
    *
    * The return value is also considered upscaled, and will be downscaled (rounding down) before returning it to the
    * Vault.
    */
function _onSwapGivenIn(
    SwapRequest memory swapRequest,
    uint256 balanceTokenIn,
    uint256 balanceTokenOut
) internal virtual returns (uint256);
```

So if you do things this way you'll find that you always get zero swap fees, because you're processing two sets of balances that have the same invariant. This shuld be caught by tests.